pkg/sink/sink_test.go

```diff
@@ -359,28 +349,259 @@ func Test_HandleEvent(t *testing.T) {
 	ts := httptest.NewServer(http.HandlerFunc(r.HandleEvent))
 	defer ts.Close()
 
-	numRequests := 0
 	restClient.Client = fakerestclient.CreateHTTPClient(func(request *http.Request) (*http.Response, error) {
-		numRequests += 1
-		if diff := cmp.Diff(wantPipelineResourceURLPath, request.URL.Path); diff != "" {
-			t.Errorf("Diff request uri: -want +got: %s", diff)
-		}
-		body, err := ioutil.ReadAll(request.Body)
-		if err != nil {
-			return nil, err
-		}
-		if diff := cmp.Diff(string(wantPipelineResourceBytes), string(body)); diff != "" {
-			t.Errorf("Diff request body: -want +got: %s", diff)
-		}
 		return &http.Response{StatusCode: http.StatusCreated, Body: ioutil.NopCloser(bytes.NewReader([]byte{}))}, nil
 	})
 
-	_, err = http.Post(ts.URL, "application/json", bytes.NewReader(eventBody))
+	resp, err := http.Post(ts.URL, "application/json", bytes.NewReader(eventBody))
 	if err != nil {
 		t.Fatalf("Error creating Post request: %s", err)
 	}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatalf("Error reading response body: %s", err)
+		return
+	}
+	wantPayload := `EventListener: my-eventlistener in Namespace: foo handling event with payload: {"head_commit": {"id": "testrevision"}, "repository": {"url": "testurl"}}`
+
+	if !strings.Contains(string(body), wantPayload) {
+		t.Errorf("Diff response body: %s, should have: %s", string(body), wantPayload)
+	}
+}
+
+func TestResource_validateEvent(t *testing.T) {
```

Comment:
I wonder if we should also be calling the [`TaskRun` validation](https://github.com/tektoncd/pipeline/blob/master/pkg/apis/pipeline/v1alpha1/taskrun_validation.go#L37) rather than just using reactor mocks.